src/main/java/BasicMAPF/Solvers/PIBT/PIBT_Solver.java

```diff
@@ -158,13 +178,46 @@ protected Solution runAlgorithm(MAPF_Instance instance, RunParameters parameters
                 Map.Entry<Agent, Double> maxEntry = getMaxEntry(this.priorities); // <agent, priority> pair with max priority
                 Agent cur = maxEntry.getKey(); // agent with the highest priority
                 solvePIBT(cur, null);
+                if (this.agentCantMove) {
+                    this.agentCantMove = false;
+                    return createSolution();
+                }
             }
             updatePriorities(instance);
         }
 
         return makeSolution();
     }
 
+    /**
+     * this function called when the algorithm return null: timeout, loop, ...
+     * when this happened - the function return null if returnPartialSolutions is false.
+     * if returnPartialSolutions is true, the function will return partial solution.
+     * @return Solution - partial solution
+     */
+    @Nullable
+    private Solution createSolution() {
```

Comment:
If you can, combine this method with `makeSolution`. If you can't at least rename it to better indicate what it's for